tests/phpunit/tests/option/wpPrimeOptionCaches.php

```diff
@@ -126,5 +160,91 @@ public function test_wp_prime_option_caches_handles_empty_notoptions_cache() {
 		$notoptions = wp_cache_get( 'notoptions', 'options' );
 		$this->assertIsArray( $notoptions, 'The notoptions cache should be an array.' );
 		$this->assertArrayHasKey( 'nonexistent_option', $notoptions, 'nonexistent_option was not added to notoptions.' );
+
+		// Check getting and re-priming the options does not result in additional database queries.
+		$initial_query_count = get_num_queries();
+
+		get_option( 'nonexistent_option' );
+		$this->assertSame(
+			0,
+			get_num_queries() - $initial_query_count,
+			'Additional database queries were made getting nonexistent_option.'
+		);
+
+		wp_prime_option_caches( array( 'nonexistent_option' ) );
+		$this->assertSame(
+			0,
+			get_num_queries() - $initial_query_count,
+			'Additional database queries were made.'
+		);
+	}
+
+	/**
+	 * Tests that wp_prime_option_caches() handles a falsey option value.
+	 *
+	 * As options are stored in the database as either strings or serialized data,
+	 * this only checks strings and arrays that type cast to boolean false.
+	 *
+	 * @ticket 59738
+	 *
+	 * @dataProvider data_wp_prime_option_caches_handles_falsey_option_value
+	 *
+	 * @param mixed $option_value Falsey option value.
+	 */
+	public function test_wp_prime_option_caches_handles_falsey_option_value( $option_value ) {
+		update_option( 'falsey_option', $option_value, false );
+		wp_cache_delete( 'falsey_option', 'options' );
+		$options_to_prime = array( 'falsey_option' );
+
+		wp_cache_delete( 'falsey_option', 'options' );
+		$this->assertFalse( wp_cache_get( 'falsey_option', 'options' ), 'falsey_option was not deleted from the cache.' );
+
+		// Call the wp_prime_option_caches function to prime the options.
+		wp_prime_option_caches( $options_to_prime );
+
+		// Store the initial database query count.
+		$initial_query_count = get_num_queries();
+
+		// Check that options are only in the 'options' cache group.
+		foreach ( $options_to_prime as $option ) {
+			$this->assertSame(
+				wp_cache_get( $option, 'options' ),
+				get_option( $option ),
+				"$option was not primed in the 'options' cache group."
+			);
+
+			$new_notoptions = wp_cache_get( $option, 'notoptions' );
+			if ( ! is_array( $new_notoptions ) ) {
+				$new_notoptions = array();
+			}
+			$this->assertArrayNotHasKey(
+				$option,
+				$new_notoptions,
+				"$option was primed in the 'notoptions' cache."
+			);
+		}
+
+		// Call the wp_prime_option_caches function to prime the options.
+		wp_prime_option_caches( $options_to_prime );
+
+		// Ensure no additional database queries were made.
+		$this->assertSame(
+			$initial_query_count,
+			get_num_queries(),
+			'Additional database queries were made.'
+		);
+	}
+
+	/**
+	 * Data provider for test_wp_prime_option_caches_handles_falsey_option_value().
+	 *
+	 * @return array[]
+	 */
+	public function data_wp_prime_option_caches_handles_falsey_option_value() {
+		return array(
+			'empty string' => array( '' ),
+			'string(0)'    => array( '0' ),
```

Comment:
Maybe cover a `0` integer and a `0.0` float too? While the database would store those as a string too, in the cache those would in fact be different values, so worth covering them as well.
```suggestion
			'string(0)'    => array( '0' ),
			'int(0)'       => array( 0 ),
			'float(0)'     => array( 0.0 ),
```

Reply:
> Maybe cover a `0` integer and a `0.0` float too? While the database would store those as a string too, in the cache those would in fact be different values, so worth covering them as well.

I realised I was testing the wrong thing.

The fundamental of the test is: re-priming an option should not trigger an additional database query. The current bug is for falsey types, but type does not matter. In 46786030eec3a5de797b24b4a6678d3d31190e86 I've refactored the entire test to test the base problem.